src/bin/console/console.cpp

```diff
@@ -211,14 +212,13 @@ int Console::_eval_command(const CommandFunction& func, const std::string& comma
 }
 
 int Console::_eval_sql(const std::string& sql) {
-  SQLQueryPlan plan;
-  hsql::SQLParserResult parse_result;
+  _sql_pipeline = std::make_unique<SQLPipeline>(sql, _explicitly_created_transaction_context);
 
   // Measure the query parse time
   auto started = std::chrono::high_resolution_clock::now();
 
   try {
-    hsql::SQLParser::parse(sql, &parse_result);
+    _sql_pipeline->get_parsed_sql();
```

Comment:
Since you are handling the exceptions within the SqlPipeline now, I would prefer having only one try-catch block in this method instead of one per call to a _sql_pipeline method, something like this:
```
try {
    ...
    _sql_pipeline->get_parsed_sql();
    ...
    _sql_pipeline->get_query_plan();
    ...
    _sql_pipeline->get_result_table();
    ...
    ...
} catch (const std::exception& exception) {
    out(exception.what());
    _explicitly_created_transaction_context = nullptr;
    return ReturnCode::Error;
}
...
return ReturnCode::Ok;
```

Reply:
This is a good idea, because the pipeline already gives meaningful exception messages. We wouldn't have to call every step by each other though.

Reply:
True. The calling of each step would only be necessary if we wanted to keep the time measurements per step. If we condense it into only the `_sql_pipeline->get_result_table()` call, we can only have one single time measurement (which I would also be okay with).

Reply:
If we want to measure the execution time, this would be better off in the pipeline, wouldn't it?

Reply:
Good point 👍

Reply:
So additionally to storing the values, we want to store their execution time?

Reply:
Let's leave that aside for now. It might come in handy later, but for now, let's get the pipeline in.